### DISCUSSION TITLE
Many to many schema creation

### DISCUSSION THREAD

**Discussion by ghost** [adonisjs/core#2943] [Help] [answered]
Two questions regarding the creation of many-to-many schema:

1. If I want to create a many to many relationship I would need to generate migrations with `node ace make:migration` for models, say, User and Skill appending the names to the command, **and** a tertiary schema called skill_user? Like: `node ace make:migration skill_user`? I found this a bit confusing in the [docs](https://docs.adonisjs.com/guides/models/relationships#many-to-many).

2. After the schema is created, if I wanted to add the [pivot column](https://docs.adonisjs.com/guides/models/relationships#additional-pivot-columns) I would have to add the column to the `skill_user` schema, like: `table.string('proficiency')`?

Are the assumptions correct? 

------

Apart from this, I also got confused regarding [many-to-many seeding](https://github.com/adonisjs/core/discussions/2181), one answer states that we would've to create a table named, in this case, SkillsUsers to be able to seed the data. In the docs, we have the following statement:

> There is no need to create a model for the pivot table.

I'm not saying it's a contradiction (creating schema !== seeding), I'm curious if it's really necessary to create such model for this pivot table to be able to seed the data.

**Comment by thetutlage** [comment, marked as answer]
Yes your assumptions are correct and which part in the docs do you find confusing?
